> Thanks for working on this! I noticed that [CI build](https://github.com/woocommerce/woocommerce/actions/runs/13535454812/job/37826191054) is failing with a specific error:
> 
> ```
> Module not found: Error: Can't resolve '@wordpress/data' in './packages/components/text-input'
> ```

Yes correct, that is the error this PR is fixing!

> The current approach of filtering out `@wordpress` paths entirely might prevent proper resolution of WordPress dependencies that are actually needed by the components. Am I missing something?
> 
> Have you confirmed that the components in Storybook don't need access to `@wordpress` packages? The error suggests that `text-input` component requires `@wordpress/data`. 🤔

Yeah I believe its coming from this file https://github.com/woocommerce/woocommerce/blob/07c2f8cb6ad9f773bf71508851554032325ac052/plugins/woocommerce-blocks/packages/components/text-input/validated-text-input.tsx#L14 that imports from `@wordpress/data`.

The thing is that the paths are actually point to the `build-types` directory: https://github.com/woocommerce/woocommerce/blob/f617c09681748acf385decd8ed80ee6c3ecc6095/plugins/woocommerce-blocks/tsconfig.base.json#L28

Removing them is fine as it will just default to the actual package installed by the package.json.
The `paths` config is only to overwrite paths. I am actually not entirely sure why both of these got added anyway, I see @opr  added them as part of this #54380 and #54331 PR.
I don't think an explanation on why they were specifically added should stop this PR though.

--- a/plugins/woocommerce-blocks/storybook/webpack.config.js
+++ b/plugins/woocommerce-blocks/storybook/webpack.config.js
@@ -16,6 +16,7 @@ const tsConfig = require( '../tsconfig.base.json' );
 
 const aliases = Object.keys( tsConfig.compilerOptions.paths ).reduce(
 	( acc, key ) => {
+		// Filter out @wordpress/* paths to allow resolution from node_modules instead of build-types directory specified in tsconfig.
 		if ( ! key.startsWith( '@wordpress' ) ) {
 			const currentPath = tsConfig.compilerOptions.paths[ key ][ 0 ];
 			acc[ key.replace( '/*', '' ) ] = path.resolve(

